NaviGator/gnc/navigator_thrust_mapper/navigator_thrust_mapper/thruster_map.py

```diff
@@ -58,6 +83,34 @@ def __init__(self, names, positions, angles, effort_ratio, effort_limit, com=np.
         self.thruster_matrix = np.hstack(thruster_matrix)
         self.thruster_matrix_inv = np.linalg.pinv(self.thruster_matrix)  # Magical numpy psuedoinverse
 
+    @classmethod
+    def from_vrx_urdf(cls, urdf_string):
+        urdf = URDF.from_xml_string(urdf_string)
+        buff = tf2_ros.Buffer()
+        listener = tf2_ros.TransformListener(buff)  # noqa
+        names = []
+        positions = []
+        angles = []
+        limit = -1
+        ratio = -1
+        for link in urdf.links:
+            print link
+            find = link.name.find('_propeller_link')
+            if find == -1:
+              continue
+            name = link.name[:find]
+            try:
+                trans = buff.lookup_transform('base_link', link.name, rospy.Time(), rospy.Duration(10))
+            except tf2_ros.TransformException as e:
+                raise Exception(e)
+            translation = rosmsg_to_numpy(trans.transform.translation)
+            rot = rosmsg_to_numpy(trans.transform.rotation)
+            yaw = euler_from_quaternion(rot)[2]
+            names.append(name)
+            positions.append(translation[0:2])
+            angles.append(yaw)
+        return cls(names, positions, angles, generate_linear_force_to_command(ratio), (250., -100.))
```

Comment:
did you mean:
        return cls(names, positions, angles, vrx_force_to_command, (250., -100.))
?
if not then where is vrx_force_to_command called?